### ISSUE TITLE
Grad-cam 3d

### ISSUE THREAD

**Issue by PARODBE** [innat/medic-ai#55]
Hi,

Thanks you for this awsome library. Do you think that It could be possible incorporate one function for grad-cam 3d. In tensorflow, you have this repo: https://www.kaggle.com/code/mariuszwisniewski/3d-grad-cam-on-ct-scans-poc

And in pytorch: https://pub.towardsai.net/explainability-for-3dresn-dc74e82044d6

Thanks!
Pablo

**Comment by innat** [comment]
Hello. Thaks for brining this feature up. Though at the moment there is no immediate plan for visualizaiton toolkit to meet other commitments. But this is indeed a good feature. If you're opne for contribution, pleae let me know. Thanks. 

Exapected features:
- [ ] GradCAM for 2D and 3D.
- [ ] Supposed to work on classification and segmentation task.
- [ ] Implemented backend-level gradient computation, a [PoC ref](https://x.com/fchollet/status/1768785771325280617).

High-levle APIs might look like this:

```python
import medicai
from medicai.utils import GradCAM

model = medicai.models.DenseNet121(...)
gcam = GradCAM(model, target_layer)
result = gcam(input_tensor)
```

Expected to have this in release `0.0.5`.

**Comment by PARODBE** [comment]
Ok! Thanks you for your response. By the way, the 3d models are not pretrained, right?

**Comment by innat** [comment]
Currently, no pretrained weights are available. Most existing pretrained weights are trained on general-purpose datasets such as ImageNet or Kinetics, which may be less effective for medical imaging tasks. However, this could change in the future.
